src/chapter_1.md

```diff
@@ -1 +1,409 @@
+```
+
+### Add to the app-agent
+
+To add to the app-agent and so use the plugin, several steps are needed.
+
+#### Add the plugin to the cargo.toml
+
+It's very easy, just add your plugin library with a matching version in the toml file:
+
+```toml
+...
+[dependencies]
+...
+my-plugin = {version= "0.1.0", path = "../my-plugin"}
+```
+
+#### Import in the app-agent
+
+In the app-agent main file, import using use:
+
+```rust,ignore
+use my_plugin::ThePlugin;
+```
+
+And then add this newly imported plugin to the statics_plugins macro:
+
+```rust,ignore
+let plugins = static_plugins![ThePlugin, CsvPlugin, SocketControlPlugin];
+```
+
+In this example, we have 3 plugins used by the app-agent:
+
+- ThePlugin
+- CsvPlugin
+- SocketControlPlugin
+
+You can now build ALUMet.
+
+## Final code
+
+```rust,ignore
+use alumet::{
+    measurement::{MeasurementAccumulator, MeasurementPoint, Timestamp},
+    metrics::TypedMetricId,
+    pipeline::{trigger::TriggerSpec, PollError, Source},
+    plugin::{
+        rust::{deserialize_config, serialize_config, AlumetPlugin},
+        ConfigTable,
+    },
+    resources::{Resource, ResourceConsumer},
+    units::{PrefixedUnit, Unit, UnitPrefix},
+};
+use serde::{Deserialize, Serialize};
+use std::{fs::File, io::Read, time::Duration};
+
+pub struct ThePlugin {
+    config: Config,
+    metrics: Option<Metrics>,
+}
+
+pub struct Metrics {
+    a_metric: TypedMetricId<u64>,
+}
+#[derive(Serialize, Deserialize, Debug)]
+struct Config {
+    #[serde(with = "humantime_serde")]
+    poll_interval: Duration,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            poll_interval: Duration::from_secs(1),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct ThePluginSource {
+    random_byte: TypedMetricId<u64>,
+}
+
+impl AlumetPlugin for ThePlugin {
+    fn name() -> &'static str {
+        "ThePlugin"
+    }
+
+    fn version() -> &'static str {
+        env!("CARGO_PKG_VERSION")
+    }
+
+    fn default_config() -> anyhow::Result<Option<ConfigTable>> {
+        Ok(Some(serialize_config(Config::default())?))
+    }
+
+    fn init(config: ConfigTable) -> anyhow::Result<Box<Self>> {
+        let config = deserialize_config(config)?;
+        Ok(Box::new(ThePlugin { config, metrics: None }))
+    }
+
+    fn start(&mut self, alumet: &mut alumet::plugin::AlumetStart) -> anyhow::Result<()> {
+        let my_byte_unit: PrefixedUnit = PrefixedUnit {
+            base_unit: Unit::Byte,
+            prefix: UnitPrefix::Plain,
+        };
+
+        let byte_metric = alumet.create_metric::<u64>("random_byte", my_byte_unit, "Byte randomly get")?;
+        self.metrics = Some(Metrics { a_metric: byte_metric });
+
+        let initial_source = Box::new(ThePluginSource {
+            random_byte: (self.metrics.as_ref().expect("Can't read byte_metric")).a_metric,
+        });
+
+        alumet.add_source(initial_source, TriggerSpec::at_interval(self.config.poll_interval));
+        Ok(())
+    }
```

Comment:
You can give this final code at the end of "Implement AlumetPlugin"

Reply:
I prefer to differentiate the AlumetPlugin part from all the others parts

Reply:
Do you know that you're implementing this function in the chapter "Implement AlumetPlugin" ?

Reply:
Yes, the first is when the tutorial do it step by step. The second time is all the code to get the full example

Reply:
But the full example is in the tutorial step by step with a todo! in start(), and then you explained what to write in start(). It may be better to put the result of the step by step guide a the end of this tutorial. As a beginner I do not want to read everything in the tutorial before finding what I'm looking for